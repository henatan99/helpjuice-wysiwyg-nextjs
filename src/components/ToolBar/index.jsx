import Image from "next/image"

const left = [
    {id: 1, content: <Image src='/assets/P.png' width='23.7px' height='27.2px' alt='P' />, margin: '0.8rem'},
    {id: 2, content: "|", margin: '0.6rem',  fontSize: '11px'},
    {id: 3, content: <Image src='/assets/clock.png' width='13.2px' height='13.2px' alt='clock' />, margin: '0.3rem'},
    {id: 4, content: "0min", margin: '0.8rem', fontSize: '14px'},
    {id: 5, content: "|", margin: '0.7rem', fontSize: '11px'},
    {id: 6, content: <Image src='/assets/rabbit.png' width='21.5px' height='21.5px' alt='rabbit' />, margin: '0.8rem'},
    {id: 7, content: "|", margin: '0.7rem',  fontSize: '11px'},
    {id: 8, content: <Image src='/assets/arrow.png' width='8px' height='8px' alt='arrow' />, margin: '0.5rem'},
    {id: 9, content: "0", margin: '1rem', fontSize: '14px'},
]

const right = [
    {id: 1, content: <Image src='/assets/check_mark.png' width='14px' height='14px' alt='check-mark' />, margin: '0.6rem'},
    {id: 2, content: <Image src='/assets/cloud.png' width='14.5px' height='10.8px' alt='cloud' />, margin: '0.5rem'},
    {id: 3, content: <Image src='/assets/more.svg' width='14px' height='14px' alt='more' />},
]

const ToolBar = () => {
    return (
        <div  style={{
            display: 'flex', 
            justifyContent: 'space-between', 
            alignItems: 'center',
            padding: '0.3rem',
            border: '0.2px solid #e5e7eb',
            borderRadius: '0.3rem',
            textAlign: 'center',
            boxShadow: '0 0.25px 0.89px 0.125px rgb(64 60 67 / 16%)'
        }}>
            <div style={{display: 'flex', alignItems: 'center'}}>
                {
                    left && left.map(elem => {
                        return (
                            <span key={elem.id} style={{marginRight: elem.margin, fontSize: elem.fontSize, textAlign: 'center', color: '#9da3ae', display: 'flex'}}>{elem.content}</span>
                        )
                    })
                }
            </div>
            <div style={{display: 'flex', alignItems: 'center'}}>
                {
                    right && right.map(elem => {
                        return (
                            <span key={elem.id} style={{marginRight: elem.margin, textAlign: 'center', color: '#9da3ae', display:'flex'}}>
                                {elem.content}
                            </span>
                        )
                    })
                }
            </div>
        </div>
    )
}

export default ToolBar