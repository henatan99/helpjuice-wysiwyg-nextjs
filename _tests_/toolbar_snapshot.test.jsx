import { render } from '@testing-library/react'
import ToolBar from '../src/components/ToolBar'

it('renders toolbar unchanged', () => {
  const { container } = render(<ToolBar />)
  expect(container).toMatchSnapshot()
})