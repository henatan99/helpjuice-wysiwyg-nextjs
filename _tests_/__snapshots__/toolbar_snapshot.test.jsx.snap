// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders toolbar unchanged 1`] = `
<div>
  <div
    style="display: flex; justify-content: space-between; align-items: center; padding: 0.3rem; border: 0.2px solid #e5e7eb; border-radius: 0.3rem; text-align: center; box-shadow: 0 0.25px 0.89px 0.125px rgb(64 60 67 / 16%);"
  >
    <div
      style="display: flex; align-items: center;"
    >
      <span
        style="margin-right: 0.8rem; text-align: center; color: rgb(157, 163, 174); display: flex;"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2723%27%20height=%2727%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="P"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </span>
      <span
        style="margin-right: 0.6rem; font-size: 11px; text-align: center; color: rgb(157, 163, 174); display: flex;"
      >
        |
      </span>
      <span
        style="margin-right: 0.3rem; text-align: center; color: rgb(157, 163, 174); display: flex;"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2713%27%20height=%2713%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="clock"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </span>
      <span
        style="margin-right: 0.8rem; font-size: 14px; text-align: center; color: rgb(157, 163, 174); display: flex;"
      >
        0min
      </span>
      <span
        style="margin-right: 0.7rem; font-size: 11px; text-align: center; color: rgb(157, 163, 174); display: flex;"
      >
        |
      </span>
      <span
        style="margin-right: 0.8rem; text-align: center; color: rgb(157, 163, 174); display: flex;"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2721%27%20height=%2721%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="rabbit"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </span>
      <span
        style="margin-right: 0.7rem; font-size: 11px; text-align: center; color: rgb(157, 163, 174); display: flex;"
      >
        |
      </span>
      <span
        style="margin-right: 0.5rem; text-align: center; color: rgb(157, 163, 174); display: flex;"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%278%27%20height=%278%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="arrow"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </span>
      <span
        style="margin-right: 1rem; font-size: 14px; text-align: center; color: rgb(157, 163, 174); display: flex;"
      >
        0
      </span>
    </div>
    <div
      style="display: flex; align-items: center;"
    >
      <span
        style="margin-right: 0.6rem; text-align: center; color: rgb(157, 163, 174); display: flex;"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2714%27%20height=%2714%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="check-mark"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </span>
      <span
        style="margin-right: 0.5rem; text-align: center; color: rgb(157, 163, 174); display: flex;"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2714%27%20height=%2710%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="cloud"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </span>
      <span
        style="text-align: center; color: rgb(157, 163, 174); display: flex;"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2714%27%20height=%2714%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="more"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </span>
    </div>
  </div>
</div>
`;
